clippy_lints/src/matches/redundant_guards.rs

```diff
@@ -177,8 +178,8 @@ fn expr_can_be_pat(cx: &LateContext<'_>, expr: &Expr<'_>) -> bool {
             ExprKind::AddrOf(..)
             | ExprKind::Array(..)
             | ExprKind::Tup(..)
-            | ExprKind::Struct(..)
-            | ExprKind::Lit(..) => true,
+            | ExprKind::Struct(..) => true,
+            ExprKind::Lit(lit) if !matches!(lit.node, LitKind::Float(..)) => true,
```

Comment:
Should probably document in `expr_can_be_pat` that it disallows this. Currently this can be a pat but won't be in future versions

Reply:
I meant in its documentation